libs/feature/search/src/lib/elasticsearch/mapper/atomic-operations.ts

```diff
@@ -0,0 +1,47 @@
+import { MetadataLink } from '@geonetwork-ui/util/shared'
+
+export type SourceWithUnknownProps = { [key: string]: unknown }
+
+export const selectField = <T>(
+  source: SourceWithUnknownProps,
+  fieldName: string
+): T | null =>
+  source !== null && fieldName in source ? (source[fieldName] as T) : null
+
+export const selectFallbackFields = <T>(
+  source: SourceWithUnknownProps,
+  ...fieldNames: string[]
+): T | null =>
+  fieldNames.reduce<T>(
+    (prev, curr) => (prev === null ? selectField(source, curr) : prev),
+    null
+  )
+
+export const selectFallback = (field, fallback) =>
+  field === null ? fallback : field
+
+export const selectTranslatedField = <T>(
```

Comment:
What value does `selectTranslatedField(source, 'title')` bring more than  `source?.title` ? The `null` set ?
Brings complexity IMO but why not

Reply:
For now, it only reads the `default` value, but I guess at some point we'll read the value from the current UI language

I agree that the added value isn't huge right now but I felt it was important to give some kind of structure to the mapping code, esp. since we explicitly support extending the mapping logic with custom fields